sdk/src/client/secret/ledger_nano.rs

```diff
@@ -244,39 +237,29 @@ impl SecretManage for LedgerSecretManager {
         let input_len = prepared_transaction.inputs_data.len();
 
         for input in &prepared_transaction.inputs_data {
-            let bip32_indices = match &input.chain {
-                Some(chain) => {
-                    chain
-                        .segments()
-                        .iter()
-                        // XXX: "ser32(i)". RTFSC: [crypto::keys::slip10::Segment::from_u32()]
-                        .map(|seg| u32::from_be_bytes(seg.bs()))
-                        .collect::<Vec<_>>()
-                }
-                None => return Err(Error::MissingBip32Chain)?,
-            };
+            let chain = input.chain.ok_or(Error::MissingBip32Chain)?;
 
             // coin_type and account_index should be the same in each output
-            if (coin_type.is_some() && coin_type != Some(bip32_indices[1]))
-                || (account_index.is_some() && account_index != Some(bip32_indices[2]))
+            if (coin_type.is_some() && coin_type != Some(chain.coin_type))
+                || (account_index.is_some() && account_index != Some(chain.account))
             {
                 return Err(Error::Bip32ChainMismatch.into());
             }
 
-            coin_type = Some(bip32_indices[1]);
-            account_index = Some(bip32_indices[2]);
+            coin_type = Some(chain.coin_type);
+            account_index = Some(chain.account);
             input_bip32_indices.push(LedgerBIP32Index {
-                bip32_change: bip32_indices[3] | Segment::HARDEN_MASK,
-                bip32_index: bip32_indices[4] | Segment::HARDEN_MASK,
+                bip32_change: chain.change.harden().into(),
+                bip32_index: chain.address_index.harden().into(),
             });
         }
 
         if coin_type.is_none() || account_index.is_none() {
             return Err(Error::NoAvailableInputsProvided)?;
         }
 
-        let coin_type = coin_type.unwrap() & !Segment::HARDEN_MASK;
-        let bip32_account = account_index.unwrap() | Segment::HARDEN_MASK;
+        let coin_type = coin_type.unwrap();
```

Comment:
Instead of those lines 257-259 and the unwraps we could basically do an `.ok_or(Error::NoAvailableInputsProvided)?` (error needs to be converted tho) on `coin_type` and `account_index`?